context.go

```diff
@@ -9,9 +9,10 @@ import (
 	ot "github.com/opentracing/opentracing-go"
 )
 
-type contextKey struct{}
+type ContextKey string
```

Comment:
Do we really need to make this type exportable? I think it would be better to keep it internal to the Instana package. For the `AddToContext` and `GetValueFromContext` functions, we could take the key as a `string` and handle the conversion to `contextKey` internally. This approach would make the code cleaner and help prevent any misuse of the type.

Reply:
I also think that since we only have two keys supported as `contextKey`, we could rewrite this in a more secure way that restricts the use of only these two keys. Currently, we can add any string type as a key, right?

Additionally, the earlier implementation uses `struct{}` as the underlying type for keys, which helps minimize memory usage since it’s embedded in the context. Given that we only have two keys, we could use `int8` or a similar type instead of string, and leverage `iota` to define each key (such as activeSpanKey and redisKey). This approach not only uses less memory but also allows for easy extension in the future if we decide to add more keys.

Reply:
Good suggestion @nithinputhenveettil 👍 . 

> Do we really need to make this type exportable?

My motive was to use this function in instrumentation packages if required in the future. 

> Given that we only have two keys, we could use int8 or a similar type instead of string, and leverage iota to define each key (such as activeSpanKey and redisKey).

Make sense👍.  I have made the type of `contextKey` to `int8` and used `iota`. The functions are changed to `private` and can be used internally.